### ISSUE TITLE
Reduced Motion: System level support

### ISSUE THREAD

**Issue by ItsJonQ** [ItsJonQ/g2#13]
Animation and transitions should be disabled anything from `@wp-g2` respecting reducedMotion configs (whether that's an app specific config, or a user machine config).

From a DX perspective, Animation/transition disabling/enabling should be invisible.

Therefore, this has to be supported at the core system level.

I have the foundations of this working:
https://github.com/ItsJonQ/g2/blob/master/packages/styles/src/create-system/createSystemElement.js#L20

For this to work, reduceMotion needs to automatically kick in for anything using...

`@wp-g2/animations`
* Entire library

`@wp-g2/styles`

* `css`
* `styled`
* `View`

`@wp-g2/components`

* `AnimatedView`
* `View`

**Comment by ItsJonQ** [comment]
Oh boy! Was able to do this with a new `useReducedMotion` hook from `@wp-g2/styles`.

https://github.com/ItsJonQ/g2/blob/master/packages/styles/src/hooks/useReducedMotion.js

![Screen Capture on 2020-08-05 at 16-48-42](https://user-images.githubusercontent.com/2322354/89462570-9e791880-d73b-11ea-899c-3bc019c090e6.gif)

It uses the `createStore` function from `@wp-g2/context` (`zustand` under the hood) to track + coordinated state in a dedicated stream.

This allows for `Animated` components to recognize app controlled `useReducedMotion` without requiring a wrapping Provider!

**Comment by ItsJonQ** [comment]
Closing this up! We have system-wide reduced motion support via the Style System. This configuration can also be controlled programatically (via JS) in addition to understanding and respecting the user's device preference.
